File: apps/api/src/xtrusio_api/schemas/workspace_stats.py
"""Pydantic schema for ``GET /api/workspaces/{workspace_id}/stats``.

A plain ``BaseModel``: the service assembles a dict of counts, there's no ORM
row to map. Every field is ``int | None`` — ``None`` means the caller lacks
the metric's permission, so the query never ran and the frontend omits the
card.
"""

from __future__ import annotations

from pydantic import BaseModel


class WorkspaceStats(BaseModel):
    """Per-metric workspace dashboard counts. ``None`` = not authorized."""

    members: int | None = None
    pending_invites: int | None = None
    recent_activity: int | None = None

File: packages/api-types/src/workspace-stats.ts
// packages/api-types/src/workspace-stats.ts
//
// Thin re-export of the generated OpenAPI WorkspaceStats schema (GET
// /api/workspaces/{workspace_id}/stats). Each field is `number | null`:
// `null` means the caller lacks that metric's permission, and the frontend
// omits its card.

import type { components } from "../generated/openapi";

export type WorkspaceStats = components["schemas"]["WorkspaceStats"];
